### DISCUSSION TITLE
Problems with Protege

### DISCUSSION THREAD

**Discussion by gregfowlerphd** [JisooSeo/101-WG-Discussions#9] [Q&A] [answered]
To familiarize myself with Protege, I recently worked through the Pizza Tutorial, available at https://tinyurl.com/NewPizzaTutorialV3-2. (Thanks to Finn and Olivia for suggesting it!) Unfortunately, I had a few problems while doing so. I'm wondering whether anyone else has experienced similar issues, can identify the source of the difficulties, or has suggestions for fixing them.

1. Page 58 of the tutorial indicates that once you have made isMilderThan the inverse of isSpicierThan and run the reasoner, the reasoner will infer that isMilderThan is transitive. But when I did this and selected isMilderThan, the box next to Transitive was not checked. (Note: It's possible that the reasoner still inferred transitivity, but without checking the box, so this might not be too much of a problem.)
2. Page 63 presents a graph created using the OntoGraf tab in Protege. Does anyone know how to use OntoGraf? When I go to the tab, the view that should contain a graph is empty. (I guess this isn't really a problem but rather a request for help.)
3. This is the **biggest problem** I had while working through the tutorial. Pages 67-71 introduce SPARQL and instruct you to execute several queries. But my SPARQL tab simply won't work. The tab is supposed to consist of two views, with the top view holding the query and bottom holding the results. Moreover, there is supposed to already be a query in the top view when you open the tab. When I open the tab, however, there is only one view (labelled "SPARQL query"), which contains no initial query and which doesn't allow you to enter one. (See the attached for a screenshot.) Any idea what's going on here?

Thanks in advance for any help you can provide!
![SPARQL Screenshot](https://github.com/JisooSeo/101-WG-Discussions/assets/158520057/d559c262-f941-4f10-b5f1-35d7ce816463)

**Comment by JisooSeo** [comment]
Thanks for the questions, Greg! I will take a look at it and try to contribute with possible answers :)

**Comment by gregfowlerphd** [comment, marked as answer]
After discussing my issue with SPARQL in the WG a couple of weeks ago, John did some digging, referring me to a discussion on Stack Overflow: https://stackoverflow.com/questions/44328191/sparql-query-tab-in-protege-doesnt-show-anything. He also put me in touch with Jon Vadja, who Carter indicated had encountered a similar problem. Unfortunately, the suggested solution--namely, to increase the heap size--didn't work.

Because it didn't, I decided to look further into the problem by Googling the beginning of the (rather long) error message from Protege's error log. Apparently, this is a known issue, as the links below indicate:

https://github.com/protegeproject/protege-distribution/issues/35
https://github.com/protegeproject/protege/issues/1110
https://github.com/protegeproject/protege/issues/625
https://github.com/protegeproject/protege/issues/662
https://stackoverflow.com/questions/75854980/sparql-query-view-in-protege-is-not-working-when-an-ontology-is-opened

Indeed, although some people in the various threads linked above claim the problem has been solved, at least one thread participant indicated that they had experienced it within the last week.

There were a couple of solutions suggested in the linked threads. The one that worked for me was to download an older version of Protege (5.5) in which the SPARQL query tab is functional. The only problem with this solution is that everything is much smaller in the older version of Protege (see the attached screenshots, the first of which is of the older version of Protege and the second of which is of the current version), to the point of being almost unreadable. So, at this point, I have both Protege 5.5 and the current version on my tablet, and I'll probably only use 5.5 for SPARQL.

If anyone has any suggestions for how to magnify the user interface in 5.5 so that it'll actually be readable, I'm all ears.
![Old Protege Screenshot](https://github.com/JisooSeo/101-WG-Discussions/assets/158520057/1c124733-e388-4e79-8612-2c64d2484ee4)
![Current Protege Screenshot](https://github.com/JisooSeo/101-WG-Discussions/assets/158520057/ced180c8-8d77-4bac-bfb8-c93d6aa19912)

**Reply by JisooSeo** [reply]
Wow, the fonts are way smaller, indeed! Would you like your comment to be the answer to the question?

**Reply by gregfowlerphd** [reply]
Good idea, I just marked it as the answer. :)
